services/web/server/tests/unit/with_dbs/02/test_projects_crud_handlers__patch.py

```diff
@@ -0,0 +1,193 @@
+# pylint: disable=protected-access
+# pylint: disable=redefined-outer-name
+# pylint: disable=too-many-arguments
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-statements
+
+
+import json
+from http import HTTPStatus
+
+import pytest
+from aiohttp.test_utils import TestClient
+from pytest_mock.plugin import MockerFixture
+from pytest_simcore.helpers.utils_assert import assert_status
+from pytest_simcore.helpers.utils_login import UserInfoDict
+from servicelib.aiohttp import status
+from simcore_service_webserver._meta import api_version_prefix
+from simcore_service_webserver.db.models import UserRole
+from simcore_service_webserver.projects.models import ProjectDict
+
+API_PREFIX = "/" + api_version_prefix
+
+
+@pytest.fixture
+def mock_catalog_api_get_services_for_user_in_product(mocker: MockerFixture):
+    mocker.patch(
+        "simcore_service_webserver.projects._crud_handlers.get_services_for_user_in_product",
+        spec=True,
+        return_value=[],
+    )
+
+
+@pytest.fixture
+def mock_project_uses_available_services(mocker: MockerFixture):
+    mocker.patch(
+        "simcore_service_webserver.projects._crud_handlers.project_uses_available_services",
+        spec=True,
+        return_value=True,
+    )
+
+
+@pytest.mark.parametrize(
+    "user_role,expected", [(UserRole.USER, status.HTTP_204_NO_CONTENT)]
+)
+async def test_patch_project(
```

Comment:
the idea of this parameterization is to check for each handler whether they follow the correct expectations based on the UserRole. Here you only check the USER one.
Unless you test elsewhere the other roles, I think you should use the fixtures available that allow to test all of the roles (unless you do it somewhere else?). We can discuss that tomorrow.